api/services/app_service.py

```diff
@@ -158,7 +158,7 @@ def get_app(self, app: App) -> App:
             model_config: AppModelConfig = app.app_model_config
             agent_mode = model_config.agent_mode_dict
             # decrypt agent tool parameters if it's secret-input
-            for tool in agent_mode.get("tools") or []:
+            for tool in agent_mode.get("tools", []):
```

Comment:
If a dict could originate from a Text or JSON field in the database, now might not be the best time to use the `dict.get(key, default)` pattern.

Reply:
cool will change that